src/main/java/com/softserveinc/dokazovi/aop/PostLogger.java

```diff
@@ -131,4 +102,35 @@ private void makeEntryInLogs(String title, UserPrincipal userPrincipal, String c
                 .build();
         logRepository.save(log);
     }
+
+    private static boolean getBooleanFromArray(Object[] arguments) {
+        return (boolean) Arrays.stream(arguments)
+                .filter(obj -> obj instanceof Boolean)
+                .findFirst().orElseThrow(() -> new NoSuchElementException("Unable find argument"));
+    }
+
+    private static Integer getPostIdFromArray(Object[] arguments) {
+        return (Integer) Arrays.stream(arguments)
+                .filter(obj -> obj instanceof Integer)
+                .findFirst().orElseThrow(() -> new NoSuchElementException("Unable find argument"));
+    }
+
+    private static Integer getPostIdFromObjectInArray(Object[] arguments) {
+        return Arrays.stream(arguments)
+                .filter(obj -> obj instanceof PostSaveFromUserDTO)
+                .map(obj -> ((PostSaveFromUserDTO) obj).getId())
+                .findFirst().orElseThrow(() -> new NoSuchElementException("Unable find argument"));
+    }
+
+    private static UserPrincipal getUserPrincipalFromArray(Object[] arguments) {
+        return (UserPrincipal) Arrays.stream(arguments)
+                .filter(obj -> obj instanceof UserPrincipal)
+                .findFirst().orElseThrow(() -> new NoSuchElementException("Unable find argument"));
+    }
+
+    private static PostSaveFromUserDTO getPostSaveFromUserDTOFromArray(Object[] arguments) {
```

Comment:
You can introduce 1 method with generic type instead of 4
`
private static <T> T getArgumentFromArrayByClassType(Object[] arguments, Class<T> clazz) {
        return (T) Arrays.stream(arguments)
                .filter(obj -> clazz.isInstance(obj))
                .findFirst().orElseThrow(() -> new NoSuchElementException("Unable to find argument"));
    }
`
and call them like this
`
UserPrincipal userPrincipal = getArgumentFromArrayByClassType(arguments, UserPrincipal.class);
        PostSaveFromUserDTO postSaveFromUserDTO = getArgumentFromArrayByClassType(arguments, PostSaveFromUserDTO.class);
`

The only method which should stay the same is `private static Integer getPostIdFromObjectInArray(Object[] arguments)`